netket/optimizer/jax/stochastic_reconfiguration.py

```diff
@@ -232,6 +254,44 @@ def compute_update(self, oks, grad, out=None):
 
         return out
 
+    def compute_update_onthefly(self, samples, grad, out=None):
+        r"""
+        Solves the SR flow equation for the parameter update ẋ.
+
+        The SR update is computed by solving the linear equation
+           Sẋ = f
+        where S is the covariance matrix of the partial derivatives
+        O_i(v_j) = ∂/∂x_i log Ψ(v_j) and f is a generalized force (the loss
+        gradient).
+
+        Args:
+            samples: An array of samples
+            grad: A pytree of the forces f
+            out: A pytree of the parameter updates that will be ignored
+        """
+
+        n_samp = samples.shape[0] * n_nodes
```

Comment:
my point essentially is that this should go inside the jit-block.
That's because it's easier for the compiler to constant-propagate things that he knows are identical.
And that cleans up the function signatures so the code is more readable.

Reply:
61b0beeb81e7ff651e3b08cb75fb476ac263b5cb